src/profiler.cpp

```diff
@@ -783,6 +788,21 @@ void Profiler::writeLog(LogLevel level, const char* message, size_t len) {
     _jfr.recordLog(level, message, len);
 }
 
+void Profiler::setTraceContext(const char* trace_id, const char* span_id) {
+    if (trace_id && span_id) {
+        char* combined = (char*)malloc(strlen(trace_id) + strlen(span_id) + 1);
+        strcpy(combined, trace_id);
+        strcat(combined, span_id);
```

Comment:
This should work too:
```suggestion
        strcpy(combined + 16, span_id);
```
`strcat` will most likely scan through the string to find the current end